examples/minimal/packages/client-react-external-wallet/src/App.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, useEffect } from "react";
+import { BaseError } from "viem";
+import {
+  usePublicClient,
+  useWalletClient,
+  useAccount,
+  useConnect,
+  useDisconnect,
+  useNetwork,
+  useSwitchNetwork,
+} from "wagmi";
+import { share } from "rxjs";
+import { useNetworkConfig, useWorldContract } from "./mud/networkConfig";
+import { syncStore, StoreProvider, useStore, type Store } from "./mud/store";
+import mudConfig from "contracts/mud.config";
+
+export const App = () => {
+  const networkConfig = useNetworkConfig();
+  const publicClient = usePublicClient();
+  const { isConnected } = useAccount();
+
+  const [store, setStore] = useState<Store | null>(null);
+
+  useEffect(() => {
+    syncStore(networkConfig, publicClient).then(setStore);
+  }, []);
+
+  return (
+    <>
+      <Connect />
+      {isConnected && <Wallet />}
+      {store && (
+        <StoreProvider store={store}>
+          <Synced />
+        </StoreProvider>
+      )}
+    </>
+  );
+};
+
+export const Synced = () => {
+  const store = useStore();
+  const { chain } = useNetwork();
+  const networkConfig = useNetworkConfig();
+  const { data: walletClient } = useWalletClient();
+
+  const counter = store.useStore((state) => state.getValue(store.tables.CounterTable, {}));
+
+  return (
+    <>
+      <div>Counter: {counter?.value ?? "unset"}</div>
+      <div>{walletClient && chain && chain!.id === networkConfig.chainId && <Connected />}</div>
+    </>
+  );
+};
+
+const Connected = () => {
+  const store = useStore();
+  const {
+    worldContract: { worldContract, write$ },
+    publicClient,
+    walletClient,
+  } = useWorldContract();
+
+  useEffect(() => {
+    if (import.meta.env.DEV) {
+      import("@latticexyz/dev-tools").then(({ mount }) =>
+        mount({
```

Comment:
We'll probably have to rethink how Dev Tools mounts if the wallet client is mutable now.

You can see this in your demo video, where the Dev Tools address no longer matches the connected address:
<img width="720" alt="image" src="https://github.com/latticexyz/mud/assets/508855/0106e4a2-4f69-4844-bd0e-8537ae1e994b">

Reply:
I didn't notice that issue, but you're right. I'll look into it.